README.md

```diff
@@ -240,6 +240,7 @@ Option                                  | Description
 `install_command`                       | Command the user can run to install the `required_executable` (or alternately the specified `required_libraries`). This is intended for documentation purposes, as Overcommit does not install software on your behalf since there are too many edge cases where such behavior would result in incorrectly configured installations (e.g. installing a Python package in the global package space instead of in a virtual environment).
 `skip_file_checkout`                    | Whether to skip this hook for file checkouts (e.g. `git checkout some-ref -- file`). Only applicable to `PostCheckout` hooks.
 `skip_if`                               | Array of arguments to be executed to determine whether or not the hook should run. For example, setting this to a value of `['bash', '-c', '! which my-executable']` would allow you to skip running this hook if `my-executable` was not in the bin path.
+`ad_hoc`                                | *["Ad-hoc" line-aware command hooks](#adding-existing-line-aware-commands) only.*
```

Comment:
It seems like there should be a way to infer this without needing to be explicit. You'll also notice that we don't use the term "ad hoc" anywhere else in the project, except a few select comments. It's a bit of a lazy shortcut to referring to this concept.

It would perhaps be cleaner to simply offer high-level settings which change the behavior of the hook automatically. For example, if you specify `message_pattern`, perhaps that's indicator enough that it's an ad hoc hook.

Alternatively, perhaps we need only warn on when settings are specified which conflict. For example, specifying `message_pattern` when a hook file matching the name of the hook already exists.

Have you considered this approach? What are your thoughts?

Reply:
> It seems like there should be a way to infer this without needing to be explicit.

I think so too, I just wanted a clear separation from the existing stuff to get me started on writing the new code (I am still kind of a Ruby rookie, esp. WRT rspec).

> You'll also notice that we don't use the term "ad hoc" anywhere else in the project, except a few select comments. It's a bit of a lazy shortcut to referring to this concept.

Yeah, I noticed that! I wasn't sure why, and now I am not sure if you mean that the term "ad-hoc" should be avoided in general. Having a few precise terms help writing the doc a tad bit easier though (less repetitions/paraphrasing)...

> It would perhaps be cleaner to simply offer high-level settings which change the behavior of the hook automatically. For example, if you specify `message_pattern`, perhaps that's indicator enough that it's an ad hoc hook.

Agreed, but see further for another candidate option.

> Alternatively, perhaps we need only warn on when settings are specified which conflict. For example, specifying `message_pattern` when a hook file matching the name of the hook already exists.

Good idea, we could that in addition to the previous idea (flattening).
 
> Have you considered this approach? What are your thoughts?

I agree with everything you wrote, the fact that an empty option does mean something is reason enough to rethink the format!